### ISSUE TITLE
(Please Read and Test) "Video not available" showing as title and video

### ISSUE THREAD

**Issue by CZython** [TeamPiped/Piped#2487]
## Testing requested (edit by kavin)

Hi everyone, I request everyone to please use https://pipedtesting.kavin.rocks/ temporarily as it would help with debugging and fixing this issue. Thanks!

In LibreTube, you can set the custom API URL to https://pipedtestingapi.kavin.rocks

## Workaround Found

use one of these instances (because they somehow work):

1. [https://piped.syncpundit.io/](https://piped.syncpundit.io/)
2. [https://piped.kozmik.studio/](https://piped.kozmik.studio/)
3. [https://piped.projectsegfau.lt/](https://piped.projectsegfau.lt/)
4. [https://piped.pfcd.me/](https://piped.pfcd.me/)
5. [https://piped.yt/](https://piped.yt/)

### Official Instance

- [X] The bug is reproducable on the [official hosted instance](http://piped.video/) or is API related.

### Describe the bug

Youtube dropped support for API version that Piped uses to load videos.
_update: some videos seem to still work_

### To Reproduce

1. open ~~any~~ official piped instance
2. click on a video
3. profit ???

### Expected behavior

Loads name and Video

### Logs/Errors

instead of video there is a message:
This content isn't available.

instead of video name there is this text:
Video Not Available

## **UPDATE**:
depending on your location different videos seem to be unavailable
source: [_verification_](https://github.com/TeamPiped/Piped/issues/2487#issuecomment-1573475673) | [_different result_](https://github.com/TeamPiped/Piped/issues/2487#issuecomment-1573498420)

**Comment by qube-03** [comment]
yup, having this happen to me too, kinda scared me at first that every video was the same

**Comment by aviallon** [comment]
![Screenshot_20230602-081436_LibreTube](https://github.com/TeamPiped/Piped/assets/7479436/40a6f303-5de1-4a85-a6ef-901ffc9c5f0d)

**Comment by qube-03** [comment]
the first video mightve worked because you have it in cache, for me no video at all works

**Comment by FireMasterK** [comment]
Some regions still appear to work - CA and FR.

**Comment by maboroshin** [comment]
You won't even get a channel feed.

**Comment by NearestWater262** [comment]
How does NewPipe do it? Can we use their method?

**Comment by Cade66** [comment]
Same here : **_The content isn't available_**, whatever video, channel. I first thought that the video/channel had changed to soon find out the issue was global.
Came here, discovered the culprit : _Youtube dropped support for API version that Piped uses to load videos._
I'm in France, by the way.

As we all I do hope the issue may be resolved, I'm totally relying on Piped ...

**Comment by radhealaniesoftware** [comment]
Im also had a same issue.
Also im trying to play my upload videos and its still works.
Here is the screenshot when i play my video:
![Screenshot_20230602-174839](https://github.com/TeamPiped/Piped/assets/86042978/57f78a0c-ab72-46fe-80c8-c924e1c3ee08)
Here is the screenshot when i play the another video:
![Screenshot_20230602-174504](https://github.com/TeamPiped/Piped/assets/86042978/bea51ba2-e465-4d73-ba0d-6c59656284af)

Thank you.

**Comment by cyberhardt** [comment]
> Im also had a same issue.
Also im trying to play my upload videos and its still works.
Here is the screenshot when i play my video:

sussy sonic

**Comment by CZython** [comment]
**Require double check!**

some videos seem to be still functioning while others are not:
- functioning video: [N++ by killingpepsi and XandoToaster in 18:23 - Summer Games Done Quick 2023](https://piped.video/watch?v=IWNi8b9WI34)
- non-functioning video: [Trine Enchanted Edition by GrosHiken in 16:43 - Summer Games Done Quick 2023](https://piped.video/watch?v=YUNtjh-qufU)

and appears to be independent of channel:
- functioning video: [Building SpaceX's Starship Super Heavy One Ring At A Time.](https://piped.video/watch?v=wciUEhOFAWg)
- non-functioning video: [Starship Launch Pad Foundation Upgraded | SpaceX Boca Chica](https://piped.video/watch?v=DOxZXhCi3gk)

_Please verify and use emojis to signal if you cat replicate._

**Comment by Cade66** [comment]
@CZython, functioning and non-functioning videos you mention bring same results here as yours, exactly.

**Comment by DanielProg39** [comment]
@CZython A bit different for me: the last video you mentioned works, others don't

**Comment by DanielProg39** [comment]
Seems that Invidious experienced something similar today: https://github.com/iv-org/invidious.io/issues/71
But now everything works as intended in all Invidious instances. Maybe they know how to deal with this issue?

**Comment by Cade66** [comment]
I have dozens of videos and channels bookmarked to Piped, cannot test them all.
Yet, besides replicating @CZython's tests, at this time all of my video **channels** linked to a Piped instance display correctly whilst no **video** does (besides those mentioned by @CZython)

**Comment by cyberhardt** [comment]
> **Require double check!**
> 
> some videos seem to be still functioning while others are not:
> 
>     * functioning video: [N++ by killingpepsi and XandoToaster in 18:23 - Summer Games Done Quick 2023](https://piped.video/watch?v=IWNi8b9WI34)
> 
>     * non-functioning video: [Trine Enchanted Edition by GrosHiken in 16:43 - Summer Games Done Quick 2023](https://piped.video/watch?v=YUNtjh-qufU)
> 
> 
> and appears to be independent of channel:
> 
>     * functioning video: [Building SpaceX's Starship Super Heavy One Ring At A Time.](https://piped.video/watch?v=wciUEhOFAWg)
> 
>     * non-functioning video: [Starship Launch Pad Foundation Upgraded | SpaceX Boca Chica](https://piped.video/watch?v=DOxZXhCi3gk)
> 
> 
> _Please verify and use emojis to signal if you can replicate._

it's the opposite for me. the ones you say don't work, are working, and vice versa.

**Comment by Cade66** [comment]
I'm mentioning Invidious instances because they faced the same issue and it seems they no longer do : https://github.com/iv-org/invidious.io/issues/71#issuecomment-1573515478

**Comment by CZython** [comment]
in hope of not DDOSing, i need to check these instances:

1. [https://piped.syncpundit.io/](https://piped.syncpundit.io/)
2. [https://piped.kozmik.studio/](https://piped.kozmik.studio/)
3. [https://piped.projectsegfau.lt/](https://piped.projectsegfau.lt/)
4. [https://piped.pfcd.me/](https://piped.pfcd.me/)
5. [https://piped.yt/](https://piped.yt/)

because they seem to work...

@Cade66 @DanielProg39  @cyberhardt

**Comment by flexagoon** [comment]
@CZython the instances you listed work fine

**Comment by Cade66** [comment]
@CZython, all 5 Piped instances you mention above display correctly, channels and videos are fully functional here.
Note : I use mainly two Piped instances which, as all others I've tested, display their homepage and channels correctly but not videos (_The content isn't available)_ : https://piped.adminforge.de/  AND https://piped.chauvet.pro/

Odd as it seems that the culprit defined as '_Youtube dropped support for API version that Piped uses to load videos_' affects some Piped instances (perhaps most) yet not all, at least not those you mention.

**Comment by CZython** [comment]
I'm staring think that the "update/dropped support" is only partial, because on some instances no videos work, while on others some videos work.

and as if by magic, these specific instances work fine.

I'll close this issue and put these instances to the top as workaround, because is obviously no longer satisfies:
"open **any** piped instance"

Or is there any reason to keep this issue open?

**Comment by Bnyro** [comment]
It makes sense to keep the issue open as long as the official instance (piped.video) is affected.

**Comment by Bnyro** [comment]
Which is still the case.

**Comment by CZython** [comment]
ok, i'll leave it open

**Comment by Cade66** [comment]
Be noted, for those who redirect embedded YouTube videos as well, that not all Piped instances have ever managed those.
Among the 5 working Piped instances mentioned by @CZython only https://piped.syncpundit.io/ accepts embedded YouTube redirected videos in my experience.

**Comment by cyberhardt** [comment]
> in hope of not DDOSing, i need a check these of instances:
> 
>     1. https://piped.syncpundit.io/
> 
>     2. https://piped.kozmik.studio/
> 
>     3. https://piped.projectsegfau.lt/
> 
>     4. https://piped.pfcd.me/
> 
>     5. https://piped.yt/
> 
> 
> because they seem to work...
> 
> @Cade66 @DanielProg39 @cyberhardt

First one is a 404 for me. the other ones work - I've tested the same video on those instances (except the first one) and on the ones I use: piped.video, piped.qdi.fi and il.ax. The ones I use aren't working, but those you provided seem to work normally.

**Comment by cyberhardt** [comment]
Yup, it seems like not all instances were hit by that error. piped.video, piped.qdi.fi and il.ax are definitely having issues. shall we make a list?

**Comment by Cade66** [comment]
Not sure making a list of working instances is pertinent:
1- Maybe instance efficiency depends of user's device location:
2- When there's such an issue as the ones some (most) instances are facing, no doubt there's barain storming going on and not only here.

The scenario seems dynamic. Good to know some instances work (thanks @CZython), perhaps others will within the coming hours.
Personally I'll use https://piped.syncpundit.io/ (though Cloudflare) because it's 100% efficient here (no 404, site, channels, videos AND embedded videos all 100% efficient) given the Piped instances I use exclusively (https://piped.adminforge.de/ AND https://piped.chauvet.pro/) do not run videos (hence embedded videos) correctly at this time.

**Comment by gandbg** [comment]
For me, some videos in the subscription feed show as "Video not available" as either the title, the thumbnail, the duration or all the them combined

![image](https://github.com/TeamPiped/Piped/assets/62039938/740bf893-0c53-4f53-b8b7-f539e4ab74e2)

All of these videos, when opened, return "Video not available" as the title, the stream, the thumbnail, the publication date and the channel name ("YouTube Viewers"). Everything else seems ok

As you can probably tell by the screenshot, I'm using watch.whatever.social as the instance

**Comment by DanielProg39** [comment]
I also had an issue in the Libretube (maybe it's in-app probem, but I'll post it here, as it's related to Piped instances) that sometimes rather than show a video a completely empty screen is shown (a black rectangle for video but no description nor suggested videos) and a message about an issue with instance, however the feed itself and feed refreshing work properly. There's also a problem when sometimes nothing works at all, neither feed, nor videos (screenshot provided). All these problems occur randomly (but most frequently in the evening) and apply to all instances, which means that changing the instance doesn't resolve the problem in most cases.

**Comment by DanielProg39** [comment]
One more thing: Piped loaded videos very slowly last days, maybe this information will be useful for tracing

**Comment by tusharhero** [comment]
all my feed has been replaced with videos from one channel.
![image](https://github.com/TeamPiped/Piped/assets/54012021/fe4a4b06-e623-494e-bcca-b29c8fdced73)

**Comment by DanielProg39** [comment]
> Not sure if its related but, all my feed has been replaced with videos from one channel.

It looked the same for me too, but now my feed looks like this:
![изображение](https://github.com/TeamPiped/Piped/assets/63293432/8ff45d52-fcda-4d0e-b582-15670bbd75e8)
And so in LibreTube app

**Comment by Cade66** [comment]
> One more thing: Piped loaded videos very slowly last days, maybe this information will be useful for tracing

Not here, never has at least with the instances I've always used  (https://piped.adminforge.de/ AND https://piped.chauvet.pro/) as well as the one I'm using now until the issue is fixed (https://piped.syncpundit.io/).

I'm not an expert, far from, but I guess it's obvious that most if not all digital issues have multifactorial sources.
Slow-down may be specific to one device. If not already done : cleaning the caches, cookies, and IndexedDB may help.
Piped instances offer preferences which are really imperative to be set to get the best rendering.
Piped instances keep user's preferences in user's LocalStorage, so if you clean your browser's IndexedDB (wiping anything which is not an extension, of the form` https+++[WEBSITE]`) you'll have to re-enter preferences or restore them if saved.

**Comment by Nirose** [comment]
> How does NewPipe do it? Can we use their method?

Piped has always been using the NewPipe extractor. https://github.com/TeamPiped/Piped#features

**Comment by NebulaBC** [comment]
My instance can load videos just fine, and for that matter it can load video titles just fine too. It’s just that the videos in the subscription feed don’t show the title until I click on them. I’m using the Yattee app, and my instance is on the latest version.

**Comment by FireMasterK** [comment]
Hi everyone, I request everyone to please use https://pipedtesting.kavin.rocks/ temporarily as it would help with debugging and fixing this issue. Thanks!

In libretube, you can set the custom API URL to https://pipedtestingapi.kavin.rocks

**Comment by CZython** [comment]
@FireMasterK just tried all videos that i remember to not work and they loaded fine.
tried on two different devices.
all new videos that i tried also loaded.

just subscriptions don't work in libretube.

**Comment by FireMasterK** [comment]
> unfortunately doesn't handle embedded videos

I'm not sure what you mean by this. If you mean embeds on other sites, they should work fine :confused:

**Comment by Cade66** [comment]
> I'm not sure what you mean by this. If you mean embeds on other sites, they should work fine confused

YouTube embedded videos, sites Iframing a YouTube video. Testing, i.e. https://embedresponsively.com/ 
I use the Redirector extension on Firefox to handle as well redirects of YouTube embedded videos.
In the above testing example on https://embedresponsively.com/ the embedded YouTube video is handled by the chosen Piped instance which leads to the Frame info =
`https://piped.syncpundit.io/embed/QILiHiTD3uc?quality=1080&bufferGoal=10&enabledCodecs=vp9&disableLBRY=true&playerAutoPlay=true&autoplay=false`

**Comment by FireMasterK** [comment]
> > I'm not sure what you mean by this. If you mean embeds on other sites, they should work fine confused
> 
> YouTube embedded videos, sites Iframing a YouTube video. Testing, i.e. https://embedresponsively.com/ I use the Redirector extension on Firefox to handle as well redirects of YouTube embedded videos. In the above testing example on https://embedresponsively.com/ the embedded YouTube video is handled by the chosen Piped instance which leads to the Frame info = `https://piped.syncpundit.io/embed/QILiHiTD3uc?quality=1080&bufferGoal=10&enabledCodecs=vp9&disableLBRY=true&playerAutoPlay=true&autoplay=false`

It works fine when I put `<iframe src='https://pipedtesting.kavin.rocks/embed/TvS1lHEQoKk' style='border:0'></iframe>` into the `Gereric Iframe` tab in https://embedresponsively.com/

**Comment by Cade66** [comment]
> It works fine when I put `<iframe src='https://pipedtesting.kavin.rocks/embed/TvS1lHEQoKk' style='border:0'></iframe>` into the `Gereric Iframe` tab in https://embedresponsively.com/

It doesn't when proceeded via the Redirector extension ... odd.

**Comment by dr460nf1r3** [comment]
> Hi everyone, I request everyone to please use https://pipedtesting.kavin.rocks/ temporarily as it would help with debugging and fixing this issue. Thanks!
> 
> In libretube, you can set the custom API URL to https://pipedtestingapi.kavin.rocks

Works here, as well.

**Comment by CatWithCode** [comment]
> Hi everyone, I request everyone to please use https://pipedtesting.kavin.rocks/ temporarily as it would help with debugging and fixing this issue. Thanks!
> 
> In libretube, you can set the custom API URL to https://pipedtestingapi.kavin.rocks

Work good!

**Comment by gandbg** [comment]
> Hi everyone, I request everyone to please use https://pipedtesting.kavin.rocks/ temporarily as it would help with debugging and fixing this issue. Thanks!
> 
> In libretube, you can set the custom API URL to https://pipedtestingapi.kavin.rocks

The view playback seem to work fine, however in the feed some channels don't get displayed even though I'm subscribed to them

https://github.com/TeamPiped/Piped/assets/62039938/05cd1f4b-44f1-4c12-a5aa-29037483c876

**Comment by flexagoon** [comment]
@FireMasterK the playback on the testing instance works fine, but it is still affected by #2489, even in the RSS feed.

**Comment by DostoevskyOnLinux** [comment]
Do the working instances offer ad-blocking?

**Comment by NebulaBC** [comment]
> Do the working instances offer ad-blocking?

Yes, it’s still working perfectly on mine.

**Comment by carnager** [comment]
Playback works fine on the testing instance, but the feed is broken. Some videos are missing and for some channels I get many old videos displayed as new ones

**Comment by n-ce** [comment]
Official instance just started working.
![chrome_screenshot_1685768120355](https://github.com/TeamPiped/Piped/assets/69345507/534ccb27-2fec-4e2e-a640-88cf77812864)

**Comment by CZython** [comment]
Can confirm that this issue has been resolved on official instance, therefor i'm closing this issue.

there are still problems with official instance:
libretube cannot play/load video
subscriber feed is still broken

but those could be separate issues.

**Comment by dr460nf1r3** [comment]
Is the latest backend commit already deployed as Docker container @FireMasterK ? I see it got updated 9 hours ago which fits the latest commit, however piped.garudalinux.org still faces the issue after updating to the latest one :thinking:

**Comment by FireMasterK** [comment]
Reopening as a lot of nodes on the official instance are still blocked.

**Comment by Bnyro** [comment]
@dr460nf1r3 Reposting from the Matrix room for Piped instance owners:
`So, if you're an instance owner and your instance is affected by the "video not available" issue:
1. Update to the latest Piped Docker / Proxy version
2. Get a different IP (by switching from IPv6 to IPv4 or vice versa) OR wait until the rate limitation ends (according to an Invidious instance owner, it was gone after shutting down the instance completely two days long)`

**Comment by Dominion0815** [comment]
same here in piped-backend docker log:

`org.schabi.newpipe.extractor.exceptions.ExtractionException: Initial player response is not valid`

**Comment by dr460nf1r3** [comment]
> Reposting from the Matrix room for Piped instance owners:

Got a link to this one for me? 

Guess I'll have to take it offline for the time being. The sweet pain of Google being able to completely shut down this whole thing for a while..

**Comment by Bnyro** [comment]
@dr460nf1r3 It's invite only, I can add you if you share your Matrix ID and instance with me!

**Comment by dr460nf1r3** [comment]
Oh sure, it's @dr460nf1r3:garudalinux.org. Thanks 😊

**Comment by DostoevskyOnLinux** [comment]
I have spun up a new instance (incredibly easy, by the way; thanks for that!) and can confirm it works as expected! Hooray!

**Comment by Cade66** [comment]
Reporting for the Piped instances I use:
https://piped.adminforge.de/ : UP 100%
https://piped.chauvet.pro/ : STILL NO VIDEO
https://piped.syncpundit.io/ : UP 100% (has never been down)

**Comment by techgamers2** [comment]
do u guys know any youtube frontend without proxy coz i m facing lottta issues the instance i use got ddos attack like for 2 days then i m now facing video not available

**Comment by MVukanichh** [comment]
https://efy.piped.pages.dev is still not working

**Comment by mongstradamus** [comment]
> Reporting for the Piped instances I use: https://piped.adminforge.de/ : UP 100% https://piped.chauvet.pro/ : STILL NO VIDEO https://piped.syncpundit.io/ : UP 100% (has never been down)

I will add my two cents on the matter with my experience with piped recently. Out of the three options  [](https://piped.adminforge.de/) worked the best for me , as its never been down and all my feeds are still working properly and are showing up. Option [](https://piped.syncpundit.io/ )for me has been mixed bad I have gotten 404 message before and not all my subscriptions from feed show up, its still occurring as of right now where not all videos from subscription show up. 

On seperate note I found pipedtesting.kavin.rocks to be ok on desktop but it doesn't work on mobile for some reason. I was trying it on IOS with safari.

**Comment by FireMasterK** [comment]
> For adminforge.de videos still show as Video not available from time to time with 1:33 video length, but clicking on the videos you can watch them just fine, the title is correct and there's no issue (and also retroactively fixes the feed.) So I don't know if it's a problem with the instance or if there's something that makes feed behave weird despite playback being fine.

Seems about right, it's working fine then! There's no issues there, existing videos in the feed will be updated when they're clicked on (or the federation network updates them).

**Comment by DostoevskyOnLinux** [comment]
> It remains to be seen whether or not this will happen again.

I hope not, but I would expect it. Google likes their ad revenue too much...

**Comment by Cade66** [comment]
About my reports, now and above, I must point out that Piped instances I've checked (those I use) have never, do not include feeds' functionality given I don't use them.

From my experience together with others' stated here, it seems that,
- a site at a given time may be functional for one user whilst not for another (or am I misunderstanding, misinterpreting our reports?);
- sites' functionality, be it because they've corrected, re-corrected or because of a random factor, appear to be down again when one might have considered that those sites, up after having been down, would stay up. Indeed, in my experience 10 minutes ago I know have (no idea about feeds):

https://piped.adminforge.de/ : UP 100%
https://piped.chauvet.pro/ : STILL NO VIDEO
https://piped.syncpundit.io/ : DOWN with a special mention : homepage appears but clicking on video, channel, even options is 100% inoperative : click is like a missed punch (void).

Simply reporting my experience. As long as at least one instance is functional I remained stunned without being annoyed. I lack knowledge to even try to understand the how of the why.

**Comment by cyberhardt** [comment]
@Cade66  adminforge works for me, while piped chauvet doesn't play videos, and syncpundit returns a 404 error

**Comment by Cade66** [comment]
> adminforge works for me, while piped chauvet doesn't play videos, and syncpundit returns a 404 error

OK. So testing again on my side, right after you given your comment followed immediately your own testing:

https://piped.adminforge.de/ : UP 100% = AS YOU
https://piped.chauvet.pro/ : STILL NO VIDEO = AS YOU
https://piped.syncpundit.io/ : UP 100% <> YOU

Why is syncpundit returning a 404 on your side but displays AND is 100% UP on my side? Are we facing randomness or is a Piped instance dependent of the user's IP? ... no idea. I should have chosen science rather than litterature in my younger years :=)

**Comment by cyberhardt** [comment]
> Why is syncpundit returning a 404 on your side but displays AND is 100% UP on my side? Are we facing randomness or is a Piped instance dependent of the user's IP? ... no idea. I should have chosen science rather than litterature in my younger years :=)

no idea either, this is all very weird. nslookup says the domain is non-existent.

**Comment by Cade66** [comment]
> no idea either, this is all very weird. nslookup says the domain is non-existent.

Here piped.syncpundit.io resolves to 104.21.234.245 ...
Have you tried restarting the browser (flushing DNS cache) ?

**Comment by cyberhardt** [comment]
> > no idea either, this is all very weird. nslookup says the domain is non-existent.
> 
> Here piped.syncpundit.io resolves to 104.21.234.245 ... Have you tried restarting the browser (flushing DNS cache) ?

even with the vpn enabled (and therefore using their dns servers) I can't access it and nslookup returns non-existent server. that IP you posted is cloudfared so I can't access it directly either.

**Comment by CZython** [comment]
did you try ```dig piped.syncpundit.io``` ? did it return real ip address, or something else?

**Comment by Cade66** [comment]
> did you try `dig piped.syncpundit.io` ? did it return real ip address, or something else?

From the urlbar dig piped.syncpundit.io opens the default search engine, DDG in my case which displays as first result:

![1](https://github.com/TeamPiped/Piped/assets/65161029/f4496404-06da-4dcf-9c57-f2935c3a4a22)

BE NOTED : 104.21.234.245 is the resolved piped.syncpundit.io as mentioned above.

**Comment by CZython** [comment]
i meant ```dig piped.syncpundit.io``` as command into the same terminal where you put nslookup.

This way we could rule out DNS issues, if it resolves in terminal/command line, if not DNS is at fault.

**Comment by Cade66** [comment]
> i meant `dig piped.syncpundit.io` as command into the same terminal where you put nslookup.
> 
> This way we could rule out DNS issues, if it resolves in terminal/command line,then is not DNS at fault.

Above my skills, knowledge. Terminal, nslookup? As I said, I'm an amateur.
- I get the site's IP with a Firefox extension called SixIndicator
- I use DNSCrypt-proxy to resolve DNS (and to blackk/white list)

**Comment by cyberhardt** [comment]
> did you try `dig piped.syncpundit.io` ? did it return real ip address, or something else?

I'm not on Linux so I don't have dig

**Comment by Cade66** [comment]
> I'm not on Linux so I don't have dig

So `dig` ia a command ... gosh, I feel dumb (hope it's only a feeling).

**Comment by cyberhardt** [comment]
> > I'm not on Linux so I don't have dig
> 
> So `dig` ia a command ... gosh, I feel dumb (hope it's only a feeling).

dig and nslookup are commands you run on the terminal, it's no big deal. if you're on windows, you can just type "nskoolup domainname.com" on cmd to get the results. same thing with dig on linux.

**Comment by CZython** [comment]
... and ```nslookup``` is the correct tool for dns diagnosis, okay...

given that ```nslookup``` returns non existent server, then the problem resides in DNS
I don't know your intentions, but your DNS likely blocks CloudFlare.

if you still want to check "piped.syncpundit.io" instance then you would need to change your preferred DNS.
I would suggest Cloudflare OneDot : 1.1.1.1 to rule out other factors.

but again, i don't know your intentions, or your threat space.

**Comment by cyberhardt** [comment]
> ... and `nslookup` is the correct tool for dns diagnosis, okay...
> 
> given that `nslookup` returns non existent server, then the problem resides in DNS I don't know your intentions, but your DNS likely blocks CloudFlare.
> 
> if you still want to check "piped.syncpundit.io" instance then you would need to change your preferred DNS. I would suggest Cloudflare OneDot : 1.1.1.1 to rule out other factors.
> 
> but again, i don't know your intentions, or your threat space.

...what do you mean by that? I used nslookup because it is the command I know to get a site's ip. I tried both with and without my vpn. I can't use dig because I'm not on linux. if you know of a better command on windows, I'll give it a go. my dns doesn't block cloudfare, otherwise I'd be having issues with the dozen other cloudfared stuff I access. I'll try with 1.1.1.1 as dns and see what happens.

**Comment by Cade66** [comment]
> if you're on windows, you can just type "nskoolup domainname.com" on cmd to get the results. same thing with dig on linux.

From cmd; `nskoolup piped.syncpundit.io` displays (Windows 7 here):
`'nskoolup' is not recognized as an internal or external command or external command, an executable program or a command file`

**Comment by CZython** [comment]
> that IP you posted is cloudfared so I can't access it directly either.
this means that the correct ip isn't blocked by firewall.
> nslookup returns non-existent server.
this means that the DNS you are connected to doesn't know "piped.syncpundit.io" and also cannot find it by connecting to root servers, so that means it is somehow blocked

**Comment by cyberhardt** [comment]
> > if you're on windows, you can just type "nskoolup domainname.com" on cmd to get the results. same thing with dig on linux.
> 
> From cmd: `nskoolup piped.syncpundit.io` displays (Windows 7 here): `'nskoolup' is not recognized as an internal or external command or external command, an executable program or a command file`

my bad, I made a typo on my comment. it is nslookup, not nskoolup

**Comment by Cade66** [comment]
> ...what do you mean by that? I used nslookup because it is the command I know to get a site's ip.

You can check a site's IP with https://dnslytics.com/
In our case : https://dnslytics.com/search?q=piped.syncpundit.io
Answer there is : _Are you looking for IPv4 address 104.21.234.244?_

**Comment by cyberhardt** [comment]
changing my dns to 1.1.1.1 kinda worked, it now says "the page isn't redirecting properly" instead of 404. whatever then, must be an issue on my end somehow.

**Comment by CZython** [comment]
> > ...what do you mean by that? I used nslookup because it is the command I know to get a site's ip.
> 
> You can check a site's IP with https://dnslytics.com/ In our case : https://dnslytics.com/search?q=piped.syncpundit.io Answer there is : _Are you looking for IPv4 address 104.21.234.244?_

you simply cannot connect to a cloudflare IPv4 directly, and expect to see a correct website, because behind that IPv4 are thousands of websites.
CloudFlare routes your https request based on the address embedded in the request itself, and by going to "https://104.21.234.244", you are telling CloudFlare that you want website of their own server.

**Comment by Cade66** [comment]
> my bad, I made a typo on my comment. it is nslookup, not nskoolup

Translated from French:

CMD: nslookup piped.syncpundit.io
```
Non-authoritative answer: Non-authoritative answer: Server: UnKnown
Address: 127.0.0.1
Name: piped.syncpundit.io
Addresses: 104.21.234.245
	  104.21.234.244
```

**Comment by Cade66** [comment]
> by going to "https://104.21.234.244", you are telling CloudFlare that you want website of their own server.

Of course. The point was only to see what DNSlytics had to say about the IP of `piped.syncpundit.io`

**Comment by DarkCat09** [comment]
Had the same problem on my instance `yt.dc09.ru`  
But after updating docker image and recreating the container, Piped doesn't load any videos:
```
piped-piped-1  | org.schabi.newpipe.extractor.exceptions.ExtractionException: Initial player response is not valid                                    piped-piped-1  |        at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor.onFetchPage(YoutubeStreamExtractor.java:872)                                                 piped-piped-1  |        at org.schabi.newpipe.extractor.Extractor.fetchPage(Extractor.java:60)      piped-piped-1  |        at org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:78)                                                 piped-piped-1  |        at org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:73)                                                 piped-piped-1  |        at org.schabi.newpipe.extractor.stream.StreamInfo.getInfo(StreamInfo.java:68)                                                 piped-piped-1  |        at me.kavin.piped.server.ServerLauncher.lambda$mainServlet$7(ServerLauncher.java:96)                                          piped-piped-1  |        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)                                          piped-piped-1  |        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)       piped-piped-1  |        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)                                  piped-piped-1  |        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)                                  piped-piped-1  |        at java.base/java.lang.Thread.run(Thread.java:833)
```

**Comment by FireMasterK** [comment]
Hi everyone, the official instance should be back entirely! Thanks, everyone for testing on the testing instance.

For instance operators, please try getting a new IP for your instance and ensure you're the latest version of the new Rust-based piped proxy. You can verify this by running `docker ps | grep 1337kavin/piped-proxy:latest`. If you're running it, you should get an output.

(untested alternative if you can't get a new IP: try shutting down your instance entirely with `docker compose down`, and starting it back up when your IP seems to no longer be rate limited - someone said this was 2 days apparently)

**Comment by starnakin** [comment]
anyone have more information about the rate limit duration ?

**Comment by FireMasterK** [comment]
> anyone have more information about the rate limit duration ?

It's been 2 days, and one of my rate-limited IPs still seems to be limited :/

**Comment by dr460nf1r3** [comment]
Same here, after taking it offline when posting here it's still up to no good 😦

**Comment by ElPepeRD** [comment]
Piped.yt doesn't work

**Comment by cyberhardt** [comment]
> Piped.yt doesn't work

yeah these errors are happening once again

**Comment by Fish25op** [comment]
https://github.com/libre-tube/LibreTube/issues/4134#issuecomment-1614611231

**Comment by MattMayuga** [comment]
![image](https://github.com/TeamPiped/Piped/assets/26103475/a32f294f-b537-438a-a72c-f831bcd1067e)
I'm experiencing this issue on piped.video.

**Comment by Fish25op** [comment]
I don't think there is a solution to these issues.

**Comment by Fish25op** [comment]
![Screenshot_20230701-113929_LibreTube Debug](https://github.com/TeamPiped/Piped/assets/136016095/8fab1ef0-7b1b-4bf0-a1f2-ea8e3eb717df)
Kavin.rocks.libre

**Comment by Fish25op** [comment]
https://github.com/LuanRT/YouTube.js

Maybe you can use this instead of NewPipe Extractor.

**Comment by Bnyro** [comment]
If you had read and understood the previous comments, you wouldn't suggest that. You can't bypass ratelimits by using a different scraper... Anyways Piped uses Java and not JavaScript at the backend!

**Comment by Fish25op** [comment]
But how will this problem be solved?

Thank you

**Comment by Moonstone3010** [comment]
I don't know if this will help, but....
Since using the main official instance gave me caching problems from the beginning, I chose another one from the official list whose server is physically closest to me. This instance recently magically disappeared from the list, but I still have it set as custom and everything works.

**Comment by Fish25op** [comment]
Great, this issue may still be the same as above

**Comment by DanielProg39** [comment]
Idk if this is relevant to this thread or not, but I recently discovered that Shorts section in the clannel page does not work

**Comment by Bellisario** [comment]
Hello, I'm the developer of [Musicale](https://github.com/Bellisario/musicale), which uses Piped's API to search and play YouTube Music songs.\
Currently my player seems not to work, even using the test instance. In fact, by filtering the results as "music only" (`music_songs`), the "items" array containing the songs is empty.

<img width="645" alt="image" src="https://github.com/TeamPiped/Piped/assets/72039923/f0a4c56b-0c7a-41b0-af59-54622c843b49">


Any solution?

Other instances seems to have the same problem and I'm unable to get results and play music...

**Comment by Moonstone3010** [comment]
> Hi everyone, I request everyone to please use https://pipedtesting.kavin.rocks/ temporarily as it would help with debugging and fixing this issue. Thanks!
> [...]
> 
> use one of these instances (because they somehow work):
> 
>     1. https://piped.syncpundit.io/
> 
>     2. https://piped.kozmik.studio/
> 
>     3. https://piped.projectsegfau.lt/
> 
>     4. https://piped.pfcd.me/
> 
>     5. https://piped.yt/


2 i 4 - down
1 i 3 - everything works
5 - Initial player response is not valid

pipedtesting.kavin.rocks - does not work and does not display any error message; loads thumbnail, does not buffer video, play button does not respond

piped.video - when I click on the thumbnail from the list it doesn't load anything

**Comment by Henry-Hiles** [comment]
> ## Testing requested (edit by kavin)
> Hi everyone, I request everyone to please use https://pipedtesting.kavin.rocks/ temporarily as it would help with debugging and fixing this issue. Thanks!
> 
> In LibreTube, you can set the custom API URL to https://pipedtestingapi.kavin.rocks
> 
> ## Workaround Found
> use one of these instances (because they somehow work):
> 
> 1. https://piped.syncpundit.io/
> 2. https://piped.kozmik.studio/
> 3. https://piped.projectsegfau.lt/
> 4. https://piped.pfcd.me/
> 5. https://piped.yt/
> 
> ### Official Instance
> * [x]  The bug is reproducable on the [official hosted instance](http://piped.video/) or is API related.
> 
> ### Describe the bug
> Youtube dropped support for API version that Piped uses to load videos. _update: some videos seem to still work_
> 
> ### To Reproduce
> 1. open ~any~ official piped instance
> 2. click on a video
> 3. profit ???
> 
> ### Expected behavior
> Loads name and Video
> 
> ### Logs/Errors
> instead of video there is a message: This content isn't available.
> 
> instead of video name there is this text: Video Not Available
> 
> ### Browser, and OS with Version.
> _No response_
> 
> ### Additional context
> ## **UPDATE**:
> depending on your location different videos seem to be unavailable source: [_verification_](https://github.com/TeamPiped/Piped/issues/2487#issuecomment-1573475673) | [_different result_](https://github.com/TeamPiped/Piped/issues/2487#issuecomment-1573498420)

Testing works fine here, I'm in Canada.

**Comment by FireMasterK** [comment]
> Hello, I'm the developer of [Musicale](https://github.com/Bellisario/musicale), which uses Piped's API to search and play YouTube Music songs. Currently my player seems not to work, even using the test instance. In fact, by filtering the results as "music only" (`music_songs`), the "items" array containing the songs is empty.
> 
> <img alt="image" width="645" src="https://user-images.githubusercontent.com/72039923/250625886-f0a4c56b-0c7a-41b0-af59-54622c843b49.png">
> Any solution?
> 
> Other instances seems to have the same problem and I'm unable to get results and play music...

This is likely unrelated, could you please open a new issue?

**Comment by Bellisario** [comment]
Done! #2628

**Comment by Dominion0815** [comment]
I solved the problem by adding http_proxy enviroment for the piped-backend container.
```

    piped:
        image: 1337kavin/piped:latest
        restart: unless-stopped
        volumes:
            - ./config/config.properties:/app/config.properties:ro
        depends_on:
            - postgres
        container_name: piped-backend
        environment:
            HTTP_PROXY: 'socks5://172.60.0.50:1080'
            HTTPS_PROXY: 'socks5://172.60.0.50:1080'
            NO_PROXY: 'localhost, 127.0.0.0/8'
```

This http_proxy is a randomized location (custom script) wireguard vpn with socks5 support.

**Comment by DanielProg39** [comment]
![image](https://github.com/TeamPiped/Piped/assets/63293432/2db835b7-a881-4b01-a9e0-b0baba8d1b39)
Instance selection list doesn't show anything and the official website infinitely loads feed and videos

**Comment by DanielProg39** [comment]
> ![image](https://user-images.githubusercontent.com/63293432/251422493-2db835b7-a881-4b01-a9e0-b0baba8d1b39.png) Instance selection list doesn't show anything and the official website infinitely loads feed and videos

Everything works now 🥳

**Comment by FireMasterK** [comment]
> I solved the problem by adding http_proxy enviroment for the piped-backend container.
> 
> ```
> 
>     piped:
>         image: 1337kavin/piped:latest
>         restart: unless-stopped
>         volumes:
>             - ./config/config.properties:/app/config.properties:ro
>         depends_on:
>             - postgres
>         container_name: piped-backend
>         environment:
>             HTTP_PROXY: 'socks5://172.60.0.50:1080'
>             HTTPS_PROXY: 'socks5://172.60.0.50:1080'
>             NO_PROXY: 'localhost, 127.0.0.0/8'
> ```
> 
> This http_proxy is a randomized location (custom script) wireguard vpn with socks5 support.

You could also configure it with `REQWEST_PROXY: socks5://127.0.0.1:1080` as a configuration value or environment variable. I've added support for this in https://github.com/TeamPiped/Piped-Backend/commit/6a2c688f501b86a2ea62fab207372abc91066bde

**Comment by DanielProg39** [comment]
> Idk if this is relevant to this thread or not, but I recently discovered that Shorts section in the clannel page does not work

Don't wanna be annoying, but should I make a separate issue for this one? It's been 4 days since I spotted it. It's not clear to me if the rate limitation is the case or if this is a Piped backend bug

**Comment by EsmailELBoBDev2** [comment]
Hi, I have same issues on my instance. I tried to use `REQWEST_PROXY` and tried to use gluetun but both cases do not work. I really hope someone can help me because I tried 2 cases and made sure it connects with `curl` but whatever inside docker it does not. example https://piped.esmailelbob.xyz/watch?v=U5lR8GdcJyo

**Comment by unstablemaple** [comment]
I have the same issue on my Piped instance, none of the aforementioned workarounds worked. Any idea how to fix this?

**Comment by EsmailELBoBDev2** [comment]
@unstablemaple I figured it out with privoxy and `REQWEST_PROXY`

**Comment by unobatbayar** [comment]
I'm having this issue as of today, first time user of piped.

**Comment by BlackLight** [comment]
I'm not sure if it's related, but as of the past ~48h I'm no longer receiving updates on my account's subscription feed either.

I haven't configured the proxy for my instance (yet), but I've kicked a bit the can down the road by using my instance for authentication and other instances for streaming.

No problem so far, as the API subscription side still worked even without the dynamic IP workaround. But that no longer seems to be the case. Relevant log traces:

```
Dec 18 15:24:36 gitter docker-compose[2462956]: nginx           | 172.21.0.1 - - [18/Dec/2023:14:24:36 +0000] "POST /webhooks/pubsub HTTP/1.0" 204 0 "-" "FeedFetcher-Google; (+http://www.google.com/feedfetcher.html)" "66.249.93.172"
Dec 18 15:24:36 gitter docker-compose[2462956]: piped-backend   | org.schabi.newpipe.extractor.exceptions.ExtractionException: Initial player response is not valid
Dec 18 15:24:36 gitter docker-compose[2462956]: piped-backend   |         at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor.onFetchPage(YoutubeStreamExtractor.java:885)
Dec 18 15:24:36 gitter docker-compose[2462956]: piped-backend   |         at org.schabi.newpipe.extractor.Extractor.fetchPage(Extractor.java:60)
Dec 18 15:24:36 gitter docker-compose[2462956]: piped-backend   |         at me.kavin.piped.server.handlers.PubSubHandlers.lambda$static$1(PubSubHandlers.java:63)
Dec 18 15:24:36 gitter docker-compose[2462956]: piped-backend   |         at java.base/java.lang.Thread.run(Thread.java:1583)
```

Has anybody being experiencing the same issue lately as well (i.e. subscriptions not updating because of errors from `/pubsub`, but everything works fine if you browse to the channel page)? Would configuring a SOCK5 proxy fix it, or is it possibly an unrelated issue?

**Comment by FireMasterK** [comment]
> I'm not sure if it's related, but as of the past ~48h I'm no longer receiving updates on my account's subscription feed either.
> 
> I haven't configured the proxy for my instance (yet), but I've kicked a bit the can down the road by using my instance for authentication and other instances for streaming.
> 
> No problem so far, as the API subscription side still worked even without the dynamic IP workaround. But that no longer seems to be the case. Relevant log traces:
> 
> ```
> 172.21.0.1 - - [18/Dec/2023:14:24:36 +0000] "POST /webhooks/pubsub HTTP/1.0" 204 0 "-" "FeedFetcher-Google; (+http://www.google.com/feedfetcher.html)" "66.249.93.172"
> org.schabi.newpipe.extractor.exceptions.ExtractionException: Initial player response is not valid
>         at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeStreamExtractor.onFetchPage(YoutubeStreamExtractor.java:885)
>         at org.schabi.newpipe.extractor.Extractor.fetchPage(Extractor.java:60)
>         at me.kavin.piped.server.handlers.PubSubHandlers.lambda$static$1(PubSubHandlers.java:63)
>         at java.base/java.lang.Thread.run(Thread.java:1583)
> ```
> 
> Has anybody being experiencing the same issue lately as well (i.e. subscriptions not updating because of errors from `/pubsub`, but everything works fine if you browse to the channel page)? Would configuring a SOCK5 proxy fix it, or is it possibly an unrelated issue?

Yes, your instance's IP blocked by YouTube, you'll need to change it, use a VPN, configure a proxy, or alternatively, if you have an ipv6 prefix, use [this](https://docs.invidious.io/ipv6-rotator/#if-you-are-running-invidious-in-docker).

**Comment by BlackLight** [comment]
@FireMasterK thanks for the input! I've managed to fix the `/pubsub` exceptions by throwing a Tor container into the mix:

```
     ytproxy:
         image: 1337kavin/ytproxy:latest
         restart: unless-stopped
         volumes:
             - ytproxy:/app/socket
         container_name: ytproxy
+        depends_on:
+            - tor
+        environment:
+            REQWEST_PROXY: socks5://piped-tor:9050
+
+    tor:
+        image: lncm/tor
+        container_name: piped-tor
+        volumes:
+            - ./data/tor/data:/etc/tor
+            - ./data/tor/data:/var/lib/tor
+            - ./data/tor/run:/var/run/tor
+        restart: on-failure
+
     piped:
         image: 1337kavin/piped:latest
         restart: unless-stopped
@@ -21,7 +53,15 @@ services:
             - ./config/config.properties:/app/config.properties:ro
         depends_on:
             - postgres
+            - tor
+        environment:
+            REQWEST_PROXY: socks5://piped-tor:9050
         container_name: piped-backend
+
```

**Comment by Deon-Trevor** [comment]
> ## Testing requested (edit by kavin)
> Hi everyone, I request everyone to please use https://pipedtesting.kavin.rocks/ temporarily as it would help with debugging and fixing this issue. Thanks!
> 
> In LibreTube, you can set the custom API URL to https://pipedtestingapi.kavin.rocks
> 
> ## Workaround Found
> use one of these instances (because they somehow work):
> 
> 1. https://piped.syncpundit.io/
> 2. https://piped.kozmik.studio/
> 3. https://piped.projectsegfau.lt/
> 4. https://piped.pfcd.me/
> 5. https://piped.yt/
> 
> ### Official Instance
> * [x]  The bug is reproducable on the [official hosted instance](http://piped.video/) or is API related.
> 
> ### Describe the bug
> Youtube dropped support for API version that Piped uses to load videos. _update: some videos seem to still work_
> 
> ### To Reproduce
> 1. open ~any~ official piped instance
> 2. click on a video
> 3. profit ???
> 
> ### Expected behavior
> Loads name and Video
> 
> ### Logs/Errors
> instead of video there is a message: This content isn't available.
> 
> instead of video name there is this text: Video Not Available
> 
> ### Browser, and OS with Version.
> _No response_
> 
> ### Additional context
> ## **UPDATE**:
> depending on your location different videos seem to be unavailable source: [_verification_](https://github.com/TeamPiped/Piped/issues/2487#issuecomment-1573475673) | [_different result_](https://github.com/TeamPiped/Piped/issues/2487#issuecomment-1573498420)

Ironically my instance: [piped.syncpundit.io](https://piped.syncpundit.io/) is now facing the same problem while others seem to have fixed it. @FireMasterK please link me to the fix, I'm dreading reading through ALL the back and forth since it's from half a year back.

**Comment by BlackLight** [comment]
@Deon-Trevor if you also started experiencing issues lately, and with error logs similar to mine (e.g. the `/pubsub` API now fails too), then it's because Google has likely extended their IP bans also to the `/pubsub` API other than the streaming stuff.

A Tor container with SOCK5 thrown in the docker-compose should be enough to circumvent it if it's the case.

**Comment by Deon-Trevor** [comment]
> @Deon-Trevor if you also started experiencing issues lately, and with error logs similar to mine (e.g. the `/pubsub` API now fails too), then it's because Google has likely extended their IP bans also to the `/pubsub` API other than the streaming stuff.
> 
> A Tor container with SOCK5 thrown in the docker-compose should be enough to circumvent it if it's the case.

Just switched IP Address and it's working fine. If you don't mind please share your docker compose config so I can implement one as well

**Comment by BlackLight** [comment]
The docker-compose diff is in my [previous post](https://github.com/TeamPiped/Piped/issues/2487#issuecomment-1862356731) - you just need to add a basic `torrc` to `data/tor/data` with SOCKS5 relay.

**Comment by Deon-Trevor** [comment]
> The docker-compose diff is in my [previous post](https://github.com/TeamPiped/Piped/issues/2487#issuecomment-1862356731) - you just need to add a basic `torrc` to `data/tor/data` with SOCKS5 relay.

Thanks a bunch mate

**Comment by FireMasterK** [comment]
@BlackLight you need to use the new Rust based proxy, you're still using the old Golang one. It was changed over a year ago in https://github.com/TeamPiped/Piped-Docker/commit/a77f887fd8f7d185e9ef60875785b509f3f37a37

You can `git pull` and run `./configure-instance.sh` again, or make the changes manually too.

**Comment by FarisZR** [comment]
> Yes, your instance's IP blocked by YouTube, you'll need to change it, use a VPN, configure a proxy, or alternatively, if you have an ipv6 prefix, use [this](https://docs.invidious.io/ipv6-rotator/#if-you-are-running-invidious-in-docker).

@FireMasterK 

I have [implemented IPv6 support in docker](https://fariszr.com/en/docker-ipv6-setup-with-propagation/) and made sure that the containers have V6 connectivity, however Piped seems to always use V4, as it always has the 'Initial player response is not valid' Error, because it's using the banned V4 address and not IPv6.

Is there a way to force piped to use V6 for its connections to YouTube?

my Piped docker setup:
https://github.com/aosus/hyper-aosus/tree/main/piped

**Comment by rderensy** [comment]
I'm interested as well. I'm having the error and `force_resolve: ipv6` fixed it for invidious, but Piped still uses IPv4

**Comment by Pamilg8** [comment]
Official stations do not have this problem. It happened again last time on the lunar.icu server. Blocking problem not solved?

**Comment by omentic** [comment]
Hi. I'm getting this error on every instance I try through the LibreTube app. Testing a couple of servers directly, it appears to be a problem through the web interface too (or at least they're not loading videos) so I'm reporting it here.

**Comment by Pamilg8** [comment]
Doesn't Piped still use the ipv6 rotator provided by Invidious?

**Comment by kekhong95** [comment]
![image](https://github.com/TeamPiped/Piped/assets/17746134/996d1b4b-b1a7-4889-9f1f-e6c8f1911238)
Hello every one : 

backend : https://apitredingplay5.ddns.net/

After completing the IPv6 setup, I noticed that my backend is defaulting to the Nginx web page instead of the intended backend page. I have reviewed the configuration steps but have not been able to identify the cause of this issue. If i set server ipv4 it will work.

I would greatly appreciate your assistance in resolving this matter.
